frontend/www/js/omegaup/components/identity/Edit.vue

```diff
@@ -0,0 +1,178 @@
+                   name="stateId"
+                   v-model="selectedState">
+                <option v-bind:value="state.code.split('-')[1]"
+                        v-for="state in countryStates"
+                        v-if="countryStates.length &gt; 0">
+                  {{ state.name }}
+                </option>
+              </select>
+            </div>
+          </div>
+          <div class="form-group">
+            <label class="col-md-4 col-sm-4 control-label"
+                 for="school">{{ T.profileSchool }}</label>
+            <div class="col-md-7 col-sm-7">
+              <input class="form-control"
+                   name="school"
+                   size="20"
+                   type="text"
+                   v-model="identity.school">
+            </div><input name="schoolId"
+                 type="hidden"
+                 v-bind:value="identity.school_id">
+          </div>
+        </div>
+        <div class="form-group pull-right">
+          <button class="btn btn-primary"
+               type="submit">{{ T.wordsSaveChanges }}</button> <button class="btn btn-secundary"
+               type="reset"
+               v-on:click="onCancel">{{ T.wordsCancel }}</button>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Watch, Prop } from 'vue-property-decorator';
+
+import { T } from '../../omegaup.js';
+import { iso3166 } from '../../../../third_party/js/iso-3166-2.js/iso3166.min.js';
+
+interface Identity {
+  name: string;
+  username: string;
+  school: string;
+  school_id: number;
+  country_id: string;
+  state_id: string;
+}
+
+interface Country {
+  country_id: string;
+  name: string;
+}
+
+interface State {
+  code: string;
+  name: string;
+}
+
+export default class IdentityEdit extends Vue {
+  @Prop() identity!: Identity;
+  @Prop() countries!: Country[];
+
+  T = T;
+  selectedCountry = this.$attrs.selectedcountry || 'MX';
+  selectedState = this.$attrs.selectedstate;
+  username = this.$attrs.username;
+
+  @Watch('selectedCountry')
+  onPropertyChanged(newContry: string, oldCountry: string) {
+    if (this.identity.country_id == newContry) {
+      this.selectedState = this.identity.state_id;
+    } else {
+      this.selectedState = this.countryStates[0].code.split('-')[1];
+    }
+  }
+
+  get groupName(): string {
+    console.log(typeof this.identity);
+    if (typeof this.identity === 'undefined') {
+      return '';
+    }
+    return `${this.identity.username.split(':')[0]}`;
+  }
+
+  get identityName(): string {
+    if (typeof this.identity === 'undefined') {
+      return '';
+    }
+    return this.identity.username.split(':')[1];
+  }
+  set identityName(username) {
+    this.identity.username = `${this.groupName}:${username}`;
+  }
+
+  get countryStates(): State[] {
+    let country = iso3166.country(this.selectedCountry);
+    let countryStates = Object.keys(country.sub).map(function(code) {
```

Comment:
hay que intentar evitar el uso de `function(...) {}` para funciones anónimas y mejor usar [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions).

```suggestion
    let countryStates = Object.keys(country.sub).map((code: string) =>
        { code: code, name: country.sub[code].name });
```

también, en todas las funciones anónimas hay que declarar los tipos de los parámetros para que no se queje amargamente.

Reply:
listo

Reply:
Ya no me marcó errores, acabo de subir el PR acá: https://github.com/omegaup/iso-3166-2.js/pull/1

Seguiré trabajando con la migración del código a TypeScript porque aún tengo algunas fallas.